components/learn/modules/ROOT/pages/write-contracts.adoc

```diff
@@ -0,0 +1,86 @@
+= Write Solidity Contracts
+
+Welcome to the exciting world of smart contract development! This guide will help get you started writing Solidity contracts, importing libraries, and compiling these into deployable Ethereum bytecode. Onwards!
+
+== Learn about Solidity
+
+We won't cover language concepts, syntax, etc. in this guide. For that, you'll want to check out the following curated content, which features great learning resources for new developers and veterans alike:
+
+ * For a general overview of how Ethereum and smart contracts work, you can't go wrong with Matt Condon's https://medium.com/@mattcondon/getting-up-to-speed-on-ethereum-63ed28821bbe[Getting Up to Speed on Ethereum]
+ * The https://solidity.readthedocs.io/en/latest/introduction-to-smart-contracts.html[official Solidity documentation] is a very good starting point for people new to the language, as are their https://solidity.readthedocs.io/en/latest/security-considerations.html[security recommendations], which nicely go over the differences between blockchains and traditional software platforms
+ * A set of https://consensys.github.io/smart-contract-best-practices/[best practices] was compiled by Consensys Dilligence: these are quite extensive, and include both https://consensys.github.io/smart-contract-best-practices/recommendations/[proven patterns] to learn from and https://consensys.github.io/smart-contract-best-practices/known_attacks/[known pitfalls] to avoid
+ * A rite of passage in the learning of Solidity is playing our very own https://solidity-05.ethernaut.openzeppelin.com/[Ethernaut] game: it'll test your understanding of both language and Ethereum concepts by having you look for subtle vulnerabilities in smart contracts as you advance through the increasingly difficult levels
+
+== Setting up a Solidity Project
+
+Simple works best: all you need to do is create directory called `contracts` and store your Solidity source files (`.sol`) there. That's it!
+
+All tools will automatically pick every contract in this directory, letting you organize your code in subdirectories (e.g. `admin`, `token`, etc.). To import a file from a different directory, simply refer to it by its relative path:
+
+```solidity
+import './admin/TransferrableAdmin.sol';
+```
+
+== Using the OpenZeppelin Contracts Library
```

Comment:
I'd reshuffle the order in this guide:

1. Create a contracts folder 
2. Create a solidity file (with content!)
3. Compile it
4. Create another file and import it
5. Import contracts